### PR TITLE
Update workflow/process documents #8565

### LINKED ISSUE #8565: Update workflow/process documents
There has been some internal restructuring as well as improvements to GitHub that necessitates updates to the workflow/process document. To name a few:
- No longer need to ask dev to update branch if outdated since GitHub provides ["Allow edits from maintainers"](https://github.com/blog/2247-improving-collaboration-with-forks) (since very long ago)
- [Multiple issue/PR templates](https://github.com/blog/2495-multiple-issue-and-pull-request-templates)
- [Label descriptions](https://github.com/blog/2505-label-improvements-emoji-descriptions-and-more) (already adopted but the docs need update)
- Removing local test
- Removing new contributor intro
- Renaming `s.ToMerge` -> `s.FinalReview` and `s.MergeApproved` -> `s.ToMerge`

Also, the core team structure has changed.

Will take some time this weekend to update.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Part of #8565 

This is the easier part of the issue; the ones that requires almost no discussion when reviewing.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@whipermr5 heads up, on the more difficult part I'm planning to formalize the workflow for long-lived feature branch, among few other things.

File: docs/orientation-guide.md [hunk 1]
```diff
@@ -53,30 +53,12 @@ Of course we are happy to guide you if you encounter any difficulties when doing
...
-b) link to your fork<br>
-c) a screenshot of the change you did (as seen on the UI)<br>
-d) your real name and a short self-intro
-
-### **Phase E**: Start contributing
-
-After you complete the above tasks, you are ready to become a contributor.
-
-Steps for fixing an issue is explained in the [process document](process.md). New contributors are advised to start doing an issue labeled [d.FirstTimers](https://github.com/TEAMMATES/teammates/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3Ad.FirstTimers) (but do not do more than one of those), then to other issues labeled as [d.Contributors](https://github.com/TEAMMATES/teammates/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3Ad.Contributors).
+When you are ready for a real contribution, you are advised to start with an issue labelled [d.FirstTimers](https://github.com/TEAMMATES/teammates/issues?q=is:issue+is:open+label:d.FirstTimers) (but do not do more than one of those), then to other issues labelled [d.Contributors](https://github.com/TEAMMATES/teammates/issues?q=is:issue+is:open+label:d.Contributors).
```
Comment (whipermr5) [inline docs/orientation-guide.md]: ... then **move on** to other issues...

File: docs/orientation-guide.md [hunk 2]
```diff
@@ -53,30 +53,12 @@ Of course we are happy to guide you if you encounter any difficulties when doing
 1. [Optional but recommended] [Set up the TEAMMATES project in an IDE](ide-setup.md).
 1. Get the test to pass (more info in [_Development Guidelines_ document](development.md)). It is OK to proceed to the next phase if you have fewer than 5 failing test cases.
 
-### **Phase C**: Tinker with the code
+### **Phase C**: Learn and start contributing
 
-> You are recommended to refer to [the _Development Guidelines_ document](development.md) for this phase.
+In order to contribute effectively, you are advised to learn how to do various development routines lined out in [the _Development Guidelines_ document](development.md). Try your best to be familiar with at least building and testing the application in your local development environment.
```
Comment (whipermr5) [inline docs/orientation-guide.md]: ... how to do **the** various development routines...

File: .github/ISSUE_TEMPLATE/help-request.md [hunk 3]
```diff
@@ -1,11 +1,7 @@
-**Issue Title:** Help Request: [summary of your problem]
-
-Copy-and-paste the following template to the issue tracker and fill it in with relevant details:
-```markdown
 - **Operating System:**
 - **JDK Version:**
```
Comment (whipermr5) [inline .github/ISSUE_TEMPLATE/help-request.md]: I was thinking of adding `Google Cloud SDK Version`. What do you think?

File: docs/orientation-guide.md [hunk 4]
```diff
@@ -53,30 +53,12 @@ Of course we are happy to guide you if you encounter any difficulties when doing
...
-c) a screenshot of the change you did (as seen on the UI)<br>
-d) your real name and a short self-intro
-
-### **Phase E**: Start contributing
-
-After you complete the above tasks, you are ready to become a contributor.
-
-Steps for fixing an issue is explained in the [process document](process.md). New contributors are advised to start doing an issue labeled [d.FirstTimers](https://github.com/TEAMMATES/teammates/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3Ad.FirstTimers) (but do not do more than one of those), then to other issues labeled as [d.Contributors](https://github.com/TEAMMATES/teammates/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3Ad.Contributors).
+When you are ready for a real contribution, you are advised to start with an issue labelled [d.FirstTimers](https://github.com/TEAMMATES/teammates/issues?q=is:issue+is:open+label:d.FirstTimers) (but do not do more than one of those), then move on to other issues labelled [d.Contributors](https://github.com/TEAMMATES/teammates/issues?q=is:issue+is:open+label:d.Contributors).
+Steps for fixing an issue are lined out in the [process document](process.md).
```
Comment (damithc) [inline docs/orientation-guide.md]: Mention that if there are no first timer issues, they can look for contributor issues instead? Also, mention that they can use `e.` labels to find easier contributor issues?

**Pull request merged**